packages/eslint-plugin/configs/esnext.js

```diff
@@ -45,3 +51,16 @@ module.exports = {
 		'template-curly-spacing': [ 'error', 'always' ],
 	},
 };
+
+const result = cosmiconfigSync( 'babel' ).search();
```

Comment:
There is a known limitation in `cosmiconfig`: https://github.com/davidtheclark/cosmiconfig/issues/24. Support for `babel.config.json` is missing, so when you want to test with a custom config you need to use a different name like `babel.config.js`.

Reply:
I think you meant to link to this issue: https://github.com/davidtheclark/cosmiconfig/issues/246

This means if someone is using `babel.config.json`, eslint won't recognize it, and will use the default parser to run? Is that something common enough that it should get documented?

Reply:
> I think you meant to link to this issue: [davidtheclark/cosmiconfig#246](https://github.com/davidtheclark/cosmiconfig/issues/246)

Ooops, thanks for correcting it.
 
> This means if someone is using `babel.config.json`, eslint won't recognize it, and will use the default parser to run? Is that something common enough that it should get documented?

I will add a note in the codebase and the changelog so we can revisit that in the future but I would expect that it gets resolved in the `comsiconfig` package because there is an open PR https://github.com/davidtheclark/cosmiconfig/pull/255 that addresses it. It's also there sitting for a few months so I hope to learn what needs to happen to get in included.

Reply:
Done with https://github.com/WordPress/gutenberg/pull/37406/commits/804eee012835478e6da8f6b43ae4b947e4ea9ab4.